File: 0771-jewels-and-stones/0771-jewels-and-stones.go
func numJewelsInStones(jewels string, stones string) int {
    jewelsMap := make(map[rune]bool)
    var answer int
    for _, jewel := range jewels {
        jewelsMap[jewel] = true
    }
    for _, stone := range stones {
        if _, ok := jewelsMap[stone]; ok {
            answer++
        }
    }
    return answer
}

File: 0771-jewels-and-stones/0771-jewels-and-stones.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        HashMap<Character,Boolean> hashMap = new HashMap<>();
        int answer = 0;
        for (char jewel : jewels.toCharArray()) {
            hashMap.put(jewel, true);
        }

        for (char stone : stones.toCharArray()) {
            if (hashMap.containsKey(stone)) {
                answer++;
            }
        }
        return answer;
    }
}